src/mount.c

```diff
@@ -238,13 +239,18 @@ cc_handle_mounts(struct cc_oci_config *config, GSList *mounts)
 		struct cc_oci_mount *m = (struct cc_oci_mount *)l->data;
 
 		if (cc_oci_mount_ignore (m)) {
-			g_debug ("ignoring mount %s", m->mnt.mnt_dir);
 			continue;
 		}
 
-		g_snprintf (m->dest, sizeof (m->dest),
-				"%s%s",
-				workload_dir, m->mnt.mnt_dir);
+		if (! cc_pod_is_vm(config) && volume) {
+			g_snprintf (m->dest, sizeof (m->dest),
+				    "%s/%s/rootfs/%s", workload_dir, config->optarg_container_id, m->mnt.mnt_dir);
+		} else {
+			g_snprintf (m->dest, sizeof (m->dest),
+				    "%s%s",
```

Comment:
Missing `/` or does `workload_dir` always end in one? (if so, you could drop a slash in the format string above but maybe we should have a check in the code to ensure it ends in `/`).